### PR TITLE
Max workload cert lifetime should not apply to istiod's server cert

### PR DESCRIPTION
**Pull request by myidpt**
When using self-signed Istiod CA, Istiod server needs a long-lived cert signed by Istiod CA itself (we make its lifetime equals the CA cert lifetime). Because the Istiod CA checks the requested lifetime should be smaller than the max workload cert lifetime, we extended the max workload cert to be equal to the CA cert lifetime. This actually makes the max workload cert lifetime effectiveless.

This PR fixes this by not conducting the max workload cert lifetime check for the Istiod server's certificate issuance (called via GenKeyCertNoLifetimeChecking). The max workload cert lifetime is still effective (by default 90 days) against normal workload cert issuance.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ X ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Review by howardjohn — COMMENTED** [review]
LGTM but should there be some unit tests of this?

File: security/pkg/pki/ca/ca.go [hunk 1]
```diff
@@ -292,63 +292,26 @@ func (ca *IstioCA) Run(stopChan chan struct{}) {
...
 
 // GetCAKeyCertBundle returns the KeyCertBundle for the CA.
 func (ca *IstioCA) GetCAKeyCertBundle() util.KeyCertBundle {
 	return ca.keyCertBundle
 }
 
-// GenKeyCert() generates a certificate signed by the CA and
+// GenKeyCertNoLifetimeCheck generates a certificate signed by the CA,
 // returns the certificate chain and the private key.
-func (ca *IstioCA) GenKeyCert(hostnames []string, certTTL time.Duration) ([]byte, []byte, error) {
```
Comment (shankgan) [inline security/pkg/pki/ca/ca.go:351]: Would it be a good idea to pass in the checkLifetime parameter to this function as well? This function appears to be called by a couple of test_functions that use it to generate a key-cert pair for workload certs as well. Perhaps we just don't need lifetime checking in those cases?

File: pilot/pkg/xds/ads_test.go [hunk 2]
```diff
@@ -80,7 +80,7 @@ func TestAgent(t *testing.T) {
 	defer tearDown()
 
 	// TODO: when authz is implemented, verify labels are checked.
-	cert, key, err := bs.CA.GenKeyCert([]string{spiffe.Identity{"cluster.local", "test", "sa"}.String()}, 1*time.Hour)
+	cert, key, err := bs.CA.GenKeyCertNoLifetimeCheck([]string{spiffe.Identity{"cluster.local", "test", "sa"}.String()}, 1*time.Hour)
```
Comment (shankgan) [inline pilot/pkg/xds/ads_test.go:83]: Looks like we are generating certs for a workload. Should be okay to not check the lifetime in this case though...

**Review by shankgan — APPROVED** [review]

File: pilot/pkg/xds/ads_test.go (hunk 2 above)
Comment (myidpt) [reply pilot/pkg/xds/ads_test.go:83]: Oh, I didn't notice this test is generating for a workload... I can write a function GenKeyCert() for this test, to make this 100% accurate.

File: security/pkg/pki/ca/ca.go (hunk 1 above)
Comment (myidpt) [reply security/pkg/pki/ca/ca.go:351]: Yeah, that works. I can do that.

**Comment by myidpt** [comment]
@howardjohn I'll add a unit test for it.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**